File: bindings/ruby/examples/who.rb
require 'rbsigar'

def format_time(who)
  return Time.at(who.time).strftime("%b %e %H:%M")
end

sigar = Sigar.new

whos = sigar.who_list

whos.each do |who|
  puts who.user + "\t" + who.device + "\t" + format_time(who) + who.host
end

File: bindings/erl/examples/who.erl
#!/usr/bin/env escript

-include(sigar).

main(_) ->
    {ok, S} = sigar:start(),
    {ok, List} = sigar:who_list(S),
    lists:map(
      fun(Who) ->
              lists:map(
                fun(K) ->
                        io:format("~s\t", [sigar:get_value(K, Who)]) end,
                [user, device, host]),
              io:format("~n")
      end, List),
    sigar:stop(S).
